internal/pipeline/executor.go

```diff
@@ -207,78 +179,84 @@ func (s hydrationState) String() string {
 	return []string{"Dry", "Hydrating", "Wet"}[s]
 }
 
-//
-// hydrate hydrates a pkg and returns wet resources.
-func hydrate(p *pkg, hctx *hydrationContext) (resources []*yaml.RNode, err error) {
-	state, found := hctx.hydrationStates[p.Path()]
+// hydrate hydrates given pkg and returns wet resources.
+func hydrate(pn *pkgNode, hctx *hydrationContext) (resources []*yaml.RNode, err error) {
+	currPkg, found := hctx.pkgs[pn.Path()]
 	if found {
-		switch state {
+		switch currPkg.state {
 		case Hydrating:
-			// we have a cycle
-			// TODO(droot): improve the error to contain useful information
-			err = fmt.Errorf("found cycle in dependencies for pkg %s", p.Path())
+			// we detected a cycle
+			err = fmt.Errorf("found cycle in dependencies for pkg %s", currPkg.Path())
+			return resources, err
 		case Wet:
-			// already hydrated, we sort of assume pkgs in hctx will be populated but we
-			// can be defensive here
-			resources = hctx.pkgs[p.Path()].resources
-		default:
-			err = fmt.Errorf("invalid pkg state %v detected", state)
+			resources = currPkg.resources
+			return resources, err
+		case Dry:
+			// Is this a dead path ?
```

Comment:
?

when would the state ever be Dry?

Reply:
When package are initialized, they are in `Dry` state, but the way our recursive hydration code is implemented, we never encounter the `Dry` state so if we come here, that represent `internal error`.